src/SignalR/clients/ts/signalr/tests/HttpConnection.test.ts

```diff
@@ -795,6 +795,42 @@ describe("HttpConnection", () => {
         });
     });
 
+    it("transport handlers set before start", async () => {
+        await VerifyLogger.run(async (logger) => {
+            const availableTransport = { transport: "LongPolling", transferFormats: ["Text"] };
+            let handlersSet = false;
+
+            let httpClientGetCount = 0;
+            const options: IHttpConnectionOptions = {
+                ...commonOptions,
+                httpClient: new TestHttpClient()
+                    .on("POST", () => ({ connectionId: "42", availableTransports: [availableTransport] }))
+                    .on("GET", () => {
+                        httpClientGetCount++;
+                        if (httpClientGetCount === 1) {
+                            if ((connection as any).transport.onreceive && (connection as any).transport.onclose) {
```

Comment:
This seems like a fragile way to test if the handlers are set considering it would fail if the private "transport" property gets renamed. Even the test name of "transport handlers set before start" seems overly specific to the implementation.

Can't we verify this in some more robust way? Like by sending a message as part of the first get response and ensuring the correct `.on` handler gets called?

Reply:
> This seems like a fragile way to test if the handlers are set considering it would fail if the private "transport" property gets renamed.

Says the guy that did `(hubConnection as any).serverTimeout();` in tests!

> Even the test name of "transport handlers set before start" seems overly specific to the implementation.

Not sure what you mean here, we're literally testing the specific implementation so...

>Can't we verify this in some more robust way?

I couldn't find an easy way (read: any way) to test this without grabbing internal fields. In general we don't test race conditions unless there is a way to force the race to happen. In this case I found a way to force it but have to grab the internal fields to verify they were set. If it makes you happy I can just remove the test.

>Like by sending a message as part of the first get response and ensuring the correct .on handler gets called?

That doesn't work because the real first response is after connect returns to HttpConnection and by then it's too late as we'll be past the race.

Reply:
> Says the guy that did (hubConnection as any).serverTimeout(); in tests!

I was actually wondering if you would bring that up when I wrote this. I'm impressed you remembered 😄

Sometimes pragmatism beats purity. I didn't want to force a client test to wait for a server timeout to be exceeded just to pass. I'm a stickler for making tests run quickly, so I thought it was a good tradeoff.

If this really is too difficult to test any other way, I'm fine with it. I'm not entirely convinced that this is a hard to force race condition. You just need a transport that calls one of the callbacks prior to the start promise completes. That would probably be easiest with a mock transport inside a unit test.

Reply:
> You just need a transport that calls one of the callbacks prior to the start promise completion. That would probably be easiest with a mock transport inside a unit test.

So, we actually might want to have another test that does this, unfortunately it will hit a different code path than the main one as we check if someone passes in a custom `ITransport` and go down a different path for that case. But that doesn't help make this test nicer 😢

Reply:
I'm fine if we just test the custom ITransport in the unit test and leave this test as is.